content/blog/monorepo-plugins-migration/index.md

```diff
@@ -0,0 +1,54 @@
+
+## Improvements
+
+There are a number of improvements we introduced by moving to monorepo.
+
+# 1) Better control of dependency management.
+
+As mentioned previously, we wanted to simplify internal and third-party dependency management. Having plugins in different repositories raised concerns about having diamond dependency conflicts and challenges of having different versions of the same dependency in different repositories.
+
+Testing specific versions of a dependency is easier because it gives us the ability to test for breaking changes and backwards compatibility across the entire codebase when an update is needed. It is easier and faster to follow Backstage team updates so that we can make sure our plugins work with the latest versions of the Backstage packages.
+# 2) Better visibility of all the plugins.
+
+It is easier for contributors to test against other plugins and possibly make multiple plugin changes in a single commit or pull request. It can also help encourage more collaboration and code reuse.
+# 3) One place to store all configs and tests.
+
+We can reuse and improve CI/CD configuration and tests across all of our plugins at the same time without needing to have  separate and sometimes duplicated configuration per plugin.
+
+# 3) Easier to keep track of upstream updates.
+
+We created a workflow that runs periodically to check for updates from the Backstage team. The workflow automatically creates a pull request with updates to the package versions for all of the plugins. The workflow also runs checks to ensure everything works as expected once the changes are merged to the main branch.
+
+## Challenges
+
+The monorepo approach is not withouts its challenges. We believed we would stumble across a few, especially in terms of building and publishing packages.
+
+# 1) Build Pipelines
+
+Ensuring builds are efficient and practical is a challenge regardless of the team size or codebase. The monorepo approach results in a lot of source code in one place. We recognized that it may take more time for CI to run all required tasks in order to approve every pull request. Ultimately, we did not see a substantial increase to build time for our monorepo..
+
+# 2) Manage publishing of the packages
+
+Although all plugins are contained within a single source code repository, each plugin is individually published to https://www.npmjs.com/. We needed a tool that would allow us to publish multiple packages but also optimize the workflow to ensure only packages that have changes are published. 
```

Comment:
NPM is a well known enough service in the JS community that we don't need to link to it.

```suggestion
Although all plugins are contained within a single source code repository, each plugin is individually published to NPM. We needed a tool that would allow us to publish multiple packages but also optimize the workflow to ensure only packages that have changes are published. 
```